rdmd_test.d

```diff
@@ -48,42 +48,38 @@ else
 
 bool verbose = false;
 
-void main(string[] args)
+int main(string[] args)
 {
-    string rdmd; // path to rdmd executable
-    string defaultCompiler; // name of default compiler expected by rdmd
+    string compilers = "dmd"; // e.g. "ldmd2,gdmd" (comma-separated list of compiler names)
     bool concurrencyTest;
     string model = "64"; // build architecture for dmd
-    string testCompilerList; // e.g. "ldmd2,gdmd" (comma-separated list of compiler names)
 
     auto helpInfo = getopt(args,
-        "rdmd", "[REQUIRED] path to rdmd executable to test", &rdmd,
-        "rdmd-default-compiler", "[REQUIRED] default D compiler used by rdmd executable", &defaultCompiler,
+        "compilers", "comma-separated list of D compilers to test rdmd with (defaults to 'dmd')", &compilers,
         "concurrency", "whether to perform the concurrency test cases", &concurrencyTest,
         "m|model", "architecture to run the tests for [32 or 64]", &model,
-        "test-compilers", "comma-separated list of D compilers to test with rdmd", &testCompilerList,
         "v|verbose", "verbose output", &verbose,
     );
 
     void reportHelp(string errorMsg = null, string file = __FILE__, size_t line = __LINE__)
     {
         defaultGetoptPrinter("rdmd_test: a test suite for rdmd\n\n" ~
-                             "USAGE:\trdmd_test [OPTIONS]\n",
+                             "USAGE:\trdmd_test [OPTIONS] <rdmd-binary>\n",
                              helpInfo.options);
         enforce(errorMsg is null, errorMsg, file, line);
     }
 
-    if (helpInfo.helpWanted)
+    if (args.length == 1 || helpInfo.helpWanted)
     {
         reportHelp();
-        return;
+        return 1;
     }
-
-    if (rdmd.length == 0)
-        reportHelp("ERROR: missing required --rdmd flag");
-
-    if (defaultCompiler.length == 0)
-        reportHelp("ERROR: missing required --rdmd-default-compiler flag");
+    if (args.length > 2)
+    {
+        writefln("Error: too many non-option arguments, expected 1 but got %s", args.length - 1);
+        return 1;
```

Comment:
It would again be easier to report this error message via the `reportHelp` function (arguably it's nice to print the usage output alongside this error message and again, this avoids having to care about return values).